tests/integration/targets/podman_image/tasks/idem_push.yml

```diff
@@ -0,0 +1,445 @@
+---
+- containers.podman.podman_image:
+    name: testimage2
+    pull: false
+    push: true
+  ignore_errors: true
+  register: img_result1
+
+- name: Check outputs
+  assert:
+    that:
+      - img_result1 is failed
+      - "'Destination must be a full URL or path to a directory' in img_result1.msg"
+
+- containers.podman.podman_image:
```

Comment:
care to add another test with `name: quay.io/testing/testimage:my-tag`? That's what we're using on our side, ensuring this is working as well and won't break in the future would be nice :). I don't think you covered that use-case in the tests.
It would probably exist like img_result2 (auth required) with "my-tag" instead of "latest". This is to ensure the `:TAG` is properly detected.